### PR TITLE
W-17463948: Prevent double injection of objects in registry

### PR DESCRIPTION
**Pull request by elrodro83**

### CODE REVIEW

**Comment by elrodro83** [comment]
--validate

**Comment by elrodro83** [comment]
--validate

**Comment by elrodro83** [comment]
--validate

**Comment by elrodro83** [comment]
--validate

File: core/src/main/java/org/mule/runtime/core/internal/event/BaseEventDecorator.java [hunk 1]
```diff
@@ -181,18 +178,6 @@ public <T extends EventInternalContext> void setOperationPolicyContext(EventInte
...
 
-  /**
-   * Invoked after deserialization. This is called when the marker interface {@link DeserializationPostInitialisable} is used.
-   * This will get invoked after the object has been deserialized passing in the current MuleContext.
-   *
-   * @param muleContext the current muleContext instance
-   * @throws MuleException if there is an error initializing
-   */
-  @SuppressWarnings({"unused"})
-  private void initAfterDeserialisation(MuleContext muleContext) throws MuleException {
```
Comment (eze210) [inline core/src/main/java/org/mule/runtime/core/internal/event/BaseEventDecorator.java:192]: Do we have a test asserting that we aren't breaking something with this?

**Review by eze210 — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/mule/runtime/core/internal/event/BaseEventDecorator.java (hunk 1 above)
Comment (elrodro83) [reply core/src/main/java/org/mule/runtime/core/internal/event/BaseEventDecorator.java:192]: all existing tests that use a persistent object store

**Comment by elrodro83** [comment]
--validate

**Pull request merged**